components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/WSDL11SOAPOperationExtractor.java

```diff
@@ -0,0 +1,431 @@
+                if (WSDL_ELEMENT_NODE.equals(parentElement.getLocalName()) && parentElement.getAttributes()
+                        .getNamedItem(SOAPToRESTConstants.NAME_ATTRIBUTE).getNodeValue().equals(partElement)) {
+                    NodeList childNodes = anElemList.getChildNodes().item(1).getChildNodes();
+                    for (int j = 0; j < childNodes.getLength(); j++) {
+                        if (childNodes.item(j).getLocalName() != null && WSDL_ELEMENT_NODE
+                                .equals(childNodes.item(j).getLocalName())) {
+                            WSDLOperationParam param = new WSDLOperationParam();
+                            NamedNodeMap attributes = childNodes.item(j).getAttributes();
+                            param.setName(attributes.getNamedItem(SOAPToRESTConstants.NAME_ATTRIBUTE).getNodeValue());
+                            String dataType = attributes.getNamedItem(SOAPToRESTConstants.TYPE_ATTRIBUTE).getNodeValue()
+                                    .split(SOAPToRESTConstants.SEQUENCE_GEN.NAMESPACE_SEPARATOR)[1];
+                            Node maxOccursNode = attributes.getNamedItem(SOAPToRESTConstants.MAX_OCCURS_ATTRIBUTE);
+                            if (maxOccursNode != null && (maxOccursNode.getNodeValue().equals(SOAPToRESTConstants.UNBOUNDED)
+                                    || !maxOccursNode.getNodeValue().equals("1"))) {
+                                param.setArray(true);
+                            }
+                            if (!primitiveTypeList.contains(dataType)) {
+                                WSDLComplexType complexType = typeMap.get(dataType);
+                                if (complexType != null) {
+                                    param.setWsdlComplexType(complexType);
+                                }
+                                param.setComplexType(true);
+                            }
+                            param.setDataType(attributes.getNamedItem(SOAPToRESTConstants.TYPE_ATTRIBUTE).getNodeValue());
+                            params.add(param);
+                        }
+                    }
+                }
+            }
+        } else {
+            throw new APIMgtWSDLException("Cannot find any types from the given wsdl.");
+        }
+    }
+
+    /**
+     * gets complex types of the wsdl definition
+     *
+     * @param elemList complex type elements list from the schema
+     * @return map of the complex types
+     */
+    private Map<String, WSDLComplexType> getComplexTypeMap(List<Node> elemList) {
+        Map<String, WSDLComplexType> typeMap = new HashMap<>();
+
+        if (elemList.get(0).getAttributes() != null && elemList.get(0).getAttributes().getNamedItem(SOAPToRESTConstants.NAME_ATTRIBUTE) != null) {
+            for (Node element : elemList) {
+                if (element.getAttributes() != null && element.getAttributes().getNamedItem(SOAPToRESTConstants.NAME_ATTRIBUTE) != null) {
```

Comment:
Seems to be going beyond 120 line limit

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2